iset.mm

```diff
@@ -97573,6 +97724,256 @@ Knuth for iterated exponentiation (_Science_ 194, 1235-1242, 1976) and
     KTVJWMWKHWKAWKWQWKVLJQPVMOVNVDVOVPVQTTT $.
 
 
+$(
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+               The ` size ` (set size) function
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+$)
+
+  $c size $.
+
+  $( Extend the definition of a class to include the set size function. $)
+  csize $a class size $.
+
+  ${
+    $d x y $.
+    $( Define the set size function ` size ` , which gives the cardinality of a
+       finite set as a member of ` NN0 ` , and assigns all infinite sets the
+       value ` +oo ` .  For example, ` ( size `` { 0 , 1 , 2 } ) = 3 ` .
+
+       Note that we use the sharp sign ( ` size ` ) for this function and we
+       use the different character octothorpe ( ` # ` ) for the apartness
+       relation (see ~ df-ap ).  We adopt the former notation from Corollary
+       8.2.4 of [AczelRathjen], p. 80 (although that work only defines it for
+       finite sets).
+
+       This definition (in terms of ` U. ` and ` ~<_ ` ) is not taken directly
+       from the literature, but for finite sets should be equivalent to the
+       conventional definition that the size of a finite set is the unique
+       natural number which is equinumerous to the given set.  (Contributed by
+       Jim Kingdon, 19-Feb-2022.) $)
+    df-size $a |- size = ( ( frec (
+        ( x e. ZZ |-> ( x + 1 ) ) , 0 ) u. { <. _om , +oo >. } ) o.
+        ( x e. _V |-> U. { y e. ( _om u. { _om } ) | y ~<_ x } ) ) $.
+  $}
+
+  ${
+    $d A y z $.
+    $( The size (second half of ~ df-size ) of an infinite set is ` _om ` .
```

Comment:
"The size (second half of ~ df-size )"
Do you mean: "The ordinal size" ?

Reply:
I'm not in love with either one but "the ordinal size" seems to be the better of the two. It is kind of an intermediate lemma for something which doesn't naturally have a name, at least not a super obvious one.